src/tests/news/search.test.ts

```diff
@@ -0,0 +1,96 @@
+import request from 'supertest';
+import app from '../../index';
+
+const getSearchResultsUrl = '/api/news/search';
+
+describe('GET /api/news/search', () => {
+  describe('Validation', () => {
+    it('should validate required q parameter and return a 400 if q is missing', async () => {
+      const response = await request(app).get(getSearchResultsUrl);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe('"q" is required');
+    });
+    it('should validate required q parameter and return a 400 if q is invalid', async () => {
+      const response = await request(app)
+        .get(getSearchResultsUrl)
+        .query({ q: '' });
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe('"q" is not allowed to be empty');
+    });
+
+    it('should handle optional parameters correctly', async () => {
+      const response = await request(app)
+        .get(getSearchResultsUrl)
+        .query({ q: 'test' });
+      expect(response.status).toBe(200);
+    });
+
+    it('should return 400 for invalid date formats in from', async () => {
+      const response = await request(app)
+        .get(getSearchResultsUrl)
+        .query({ q: 'test', from: 'invalid-date-format' });
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe(
+        '"from" must be in ISO 8601 date format',
+      );
+    });
+
+    it('should return 400 for invalid date formats in to', async () => {
+      const response = await request(app)
+        .get(getSearchResultsUrl)
+        .query({ q: 'test', to: 'invalid-date-format' });
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe('"to" must be in ISO 8601 date format');
+    });
+
+    it('should return 400 for invalid sortBy value', async () => {
+      const response = await request(app).get(getSearchResultsUrl).query({
+        q: 'test',
+        sortBy: 'invalid: not relevancy, popularity, or publishedAt',
+      });
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBe(
+        '"sortBy" must be one of [relevancy, popularity, publishedAt]',
+      );
+    });
+  });
+  describe('API Response Tests', () => {
```

Comment:
I like it that you separated the validation and API response tests!

Reply:
@bojana-petroska Thanks! I agree and will add the empty lines. I am using a Jest extension in VScode to run a single test case, and it adds an extra space right above the 'it' line, so in VScode it looks like an empty line!!!